### ISSUE TITLE
Feature: adding melee hit to non-melee weapons

### ISSUE THREAD

**Issue by thecyrus** [TheGameCreators/GameGuruRepo#820]
This could be for both Classic and Max. As it's the case with most FPS's... While using a gun, enemies could come too close and maybe the player has to reload or can't shoot fast enough, I tought that we could have a button (for example the center mouse button/wheel) to use gun butt and hit the close enemy.

**Comment by mtrm92** [comment]
it exist allready i think, iam sure its in the enhacend weapon pack but not sure about the stock weapons

**Comment by fearlesswee** [comment]
This is already a feature, the Enhanced Weapons Pack guns are capable of melee strikes by pressing Left Alt. The stock weapons simply do not use this feature.

**Comment by thecyrus** [comment]
Ok, I was thinking about an implied feature for the engine itself, but I get that it's present in the Enhanced Weapons pack.

**Comment by fearlesswee** [comment]
It is an engine feature! (that's why it's even possible in the first place! GG is pretty hardcoded especially when it comes to weapons...)

I'm just saying that the Enhanced Weapons Pack is an example of weapons that advantage of this engine feature.. Sorry if I wasn't clear about that! 👍 

I guess what you're asking is to make the stock weapons also take advantage of the melee system(?)

**Comment by thecyrus** [comment]
Yes, pretty much what I meant... To make the player able to give a melee hit in every situation when a NPC is too close.
